kudosaurus.go

```diff
@@ -0,0 +1,52 @@
+package kudosaurus
+
+// Kudo - kudo entity
+type Kudo struct {
+	ID          string
+	TeamID      string
+	Text        string
+	PersonFrom  Person
+	PersonTo    Person
+	IsPublished bool
+}
+
+// Person - person entity
+type Person struct {
+	ID         string
+	Name       string
+	Kudos      []Kudo
+	KudosGiven []Kudo
+	KudosLeft  int
+	IsAdmin    bool
+}
+
+// Team - team entity
+type Team struct {
+	ID       string
+	Name     string
+	People   []Person
+	Settings Settings
+}
+
+// Settings - runtime settings for a team
+type Settings struct {
+	TeamID          string
+	TargetChannelID string
+}
+
+// Store - stores data
+type Store interface {
```

Comment:
I think it might be a good idea to separate it into few interfaces (one per each entity if possible)
Also PublishKudos() could rather be updateKudos() (from perspective of a storage layer, context of business case is irrelevant)

Reply:
Yeah, I went with this name because `update` looks a bit like something that lets you _edit_ stuff. It doesn't publish anything, either, that's for sure.